taker-frontend/src/components/Trade.tsx

```diff
@@ -93,14 +97,55 @@ const Trade = (
             const quantityAsNumber = quantity.replace("$", "");
 
             let payload: CfdOrderRequestPayload = {
-                order_id: order_id!,
+                order_id: orderId!,
                 quantity: Number.parseFloat(quantityAsNumber),
             };
             await onLongSubmit(payload);
             onClose();
         },
     });
 
+    const parse = (val: any) => Number.parseInt(val.replace(/^\$/, ""));
+
+    const balanceTooLow = walletBalance && walletBalance < parse(margin);
+    const quantityTooHigh = maxQuantity < parse(quantity);
+    const quantityTooLow = minQuantity > parse(quantity);
+    const quantityGreaterZero = parse(quantity) > 0;
+
+    const canSubmit = orderId && !isSubmitting && !balanceTooLow
+        && !quantityTooHigh && !quantityTooLow && quantityGreaterZero;
+
+    let alertBox;
+
+    if (balanceTooLow) {
+        alertBox = <Alert status="error">
+            <AlertIcon />
+            <AlertTitle mr={2}>Your balance is too low!</AlertTitle>
+            <AlertDescription>Pleas deposit more into you wallet.</AlertDescription>
+        </Alert>;
```

Comment:
nit: it might be worthwhile to extract this as a separate component (to ensure consistent styling and ease readability) - then `title` and `description` could be passed in as props.
(this can be always done later)

Reply:
done :)